### PR TITLE
add transaction.id to errors

### LINKED ISSUE #409: Connecting errors with transactions
It would be valuable if errors had a reference to the transaction that was active when the error occurred.

I suggest that we simply add an optional `transactionId` to the list of [error properties](https://github.com/elastic/apm-server/blob/master/docs/spec/errors/error.json) in the intake API. In the Elasticsearch model I imagine this should be stored as `transaction.id`.

If possible it would be nice to get this out in the data model before 6.2. We can always improve the usage of this in the UI later, but having it in the model before 6.2 would be beneficial.

Crazy bonus idea: This might even allow us to remove duplicate information from errors that are already stored on the transaction. But that would potentially have serious downsides once we introduce sampling, as the active transaction might not have been sampled.

cc @elastic/apm

### PR DESCRIPTION
**Pull request by graphaelli**
to connect errors to transactions.  closes #409

### CODE REVIEW

**Comment by roncohen** [comment]
@graphaelli nice work! 
But we should put it in the same place as in with transaction documents. That is, on the root. In other words, we need to move it one level up. That way, the field is the same across all our documents.

**Comment by graphaelli** [comment]
@roncohen that makes sense now, pushed up a commit for that.  One side-effect is that `transaction.id` is now a common field - the remaining transaction properties (`name`, `duration`, etc) stay valid only on transaction docs.

**Comment by ruflin** [comment]
As the `transaction.id` is the field for correlation, it's fine that `name` etc. are only available in the transaction event.

@graphaelli This will need a rebase.

**Comment by graphaelli** [comment]
@ruflin I was going to see if merging #443 eliminates the conflict here, do you have happen to know?

**Comment by ruflin** [comment]
I think you will have to rebase anyways even with #443 as the union is not in this PR yet and Github will not retrigger (I think).

**Comment by graphaelli** [comment]
taking your word for it, rebased & force pushed

File: tests/data/valid/error/payload.json [hunk 1]
```diff
@@ -245,5 +245,8 @@
                 "message": "Cannot read property 'baz' of undefined"
             }
         }
-    ]
+    ],
+    "transaction": {
```
Comment (roncohen) [inline tests/data/valid/error/payload.json]: I believe this should be inside the elements in `errors`. Agents can buffer up errors from different transactions

Comment (graphaelli) [reply tests/data/valid/error/payload.json]: ugh, my mistake, that makes much more sense.

File: tests/json_schema.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func TestJsonSchemaKeywordLimitation(t *testing.T, fieldPaths []string, schema s
 	mapping := []Mapping{
 		{"errors.context", "context"},
 		{"transactions.context", "context"},
+		{"errors.transaction.id", "transaction.id"},
```
Comment (graphaelli) [inline tests/json_schema.go]: @roncohen wasn't that comfortable with this, any way to do this better?

Comment (roncohen) [reply tests/json_schema.go]: Making sure i got this correctly: The test checks that everything that get stored as a `keyword` has the length restriction and vice-versa. We use this mapping to map the json schema names to the final elasticssearch document names. If that's correct, i don't have a better way to do it that what you did here.

Comment (graphaelli) [reply tests/json_schema.go]: That's how I understand that test as well, thanks for verifying.

**Pull request merged**